lib/skate/warm_up.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Skate.WarmUp do
+  use GenServer
+  require Logger
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, :ok)
+  end
+
+  @impl true
+  def init(state) do
+    pool_size = Application.get_env(:skate, Skate.Repo)[:pool_size]
+
+    minimum_percent_queries_to_succeed =
+      Application.get_env(:skate, Skate.WarmUp)[:minimum_percent_queries_to_succeed] || 1
+
+    max_attempts = Application.get_env(:skate, Skate.WarmUp)[:max_attempts]
+
+    warm_up_check_fn =
+      Application.get_env(:skate, :warm_up_test_fn, fn _index, _attemptgit ->
+        Skate.Repo.query("SELECT 1", [])
+      end)
+
+    case check_connections_warmed_up(%{
+           pool_size: pool_size,
+           required_success_count: pool_size * minimum_percent_queries_to_succeed,
+           max_attempts: max_attempts,
+           attempt: 1,
+           check_fn: warm_up_check_fn
+         }) do
+      %{status: :success} = result ->
+        Logger.info(format_result_message(result))
+        {:ok, state}
```

Comment:
**suggestion:** I think you could return `:ignore` here to cause the process to exit without affecting the rest of the supervision tree or causing it to attempt to restart, which might be more appropriate for something transient like this that's only needed at startup.

Reply:
Updated in https://github.com/mbta/skate/pull/2108/commits/72a0579f62f29e20e5f3f4e9866f677dfc9b2c5f